lib/imagec/imagec.go

```diff
@@ -325,12 +325,16 @@ func (ic *ImageC) WriteImageBlob(image *ImageWithMeta, progressOutput progress.O
 // CreateImageConfig constructs the image metadata from layers that compose the image
 func (ic *ImageC) CreateImageConfig(images []*ImageWithMeta) (metadata.ImageConfig, error) {
 
-	if len(images) == 0 {
-		return metadata.ImageConfig{}, nil
+	imageLayer := images[0] // the layer that represents the actual image
+
+	// if we already have an imageID associated with this layerID, we don't need
+	// to calculate imageID and can just grab the image config from the cache
+	id := cache.RepositoryCache().GetImageID(imageLayer.ID)
+	if image, err := cache.ImageCache().GetImage(id); err == nil {
```

Comment:
Is it possible to have an ID in the cache but no backing image in the PL?  If it is, then this check will prevent the user from rectifying the issue by issuing the `pull` again.   I'm hoping it's impossible.